src/node/kernel_notifications.cpp

```diff
@@ -57,9 +58,23 @@ static void DoWarning(const bilingual_str& warning)
 
 namespace node {
 
-void KernelNotifications::blockTip(SynchronizationState state, CBlockIndex& index)
+kernel::InterruptResult KernelNotifications::blocksImported()
+{
+    if (m_stop_after_block_import) {
+        StartShutdown();
+        return kernel::Interrupted{};
+    }
+    return {};
+}
+
+kernel::InterruptResult KernelNotifications::blockTip(SynchronizationState state, CBlockIndex& index)
 {
     uiInterface.NotifyBlockTip(state, &index);
+    if (m_stop_at_height && index.nHeight >= m_stop_at_height) {
```

Comment:
Is it possible for this to be called twice?
```c++
if (m_stop_at_height) {
  if (index.nHeight > m_stop_at_height) {
    // assert(false) or return kernel::Interrupted{} ?
  } else if (index.nHeight == m_stop_at_height) {
  ..
  }
}
```

Not that I imagine calling `StartShutdown()` twice would be particularly harmful anyway.